src/plan/barriers.rs

```diff
@@ -81,8 +79,9 @@ impl<E: ProcessEdgesWork> ObjectRememberingBarrier<E> {
                 // We just logged the object
                 return true;
             } else {
-                let old_value =
-                    load_metadata::<E::VM>(&self.meta, object, None, Some(Ordering::SeqCst));
+                let old_value = self
+                    .meta
+                    .load_atomic::<E::VM, u8>(object, None, Ordering::SeqCst);
```

Comment:
Usually cmpxchg returns the old value regardless of success or failure, so it should not be necessary to load again.  Consider changing the API of cmpxchg.

Alternatively, try `fetch_and_atomic`.  It should be faster on architectures that has special instructions that never fails.

Reply:
Yeah, I will do it in another PR.